test/boost/tablets_test.cc

```diff
@@ -1738,3 +1740,120 @@ SEASTAR_THREAD_TEST_CASE(test_load_balancing_resize_requests) {
+    snitch.invoke_on_all(&snitch_ptr::start).get();
+
+    struct ring_point {
+        double point;
+        inet_address host;
+        host_id id = host_id::create_random_id();
+    };
+
+    auto d2t = [](double d) -> int64_t {
+        // Double to unsigned long conversion will overflow if the
+        // input is greater than numeric_limits<long>::max(), so divide by two and
+        // multiply again later.
+        auto scale = std::numeric_limits<unsigned long>::max();
+        return static_cast<unsigned long>(d * static_cast<double>(scale >> 1)) << 1;
+    };
+
+    std::vector<ring_point> ring_points = {
+            { 1.0,  inet_address("192.100.10.1") },
+            { 2.0,  inet_address("192.101.10.1") },
+            { 3.0,  inet_address("192.102.10.1") },
+            { 4.0,  inet_address("192.100.20.1") },
+            { 5.0,  inet_address("192.101.20.1") },
+            { 6.0,  inet_address("192.102.20.1") },
+            { 7.0,  inet_address("192.100.30.1") },
+            { 8.0,  inet_address("192.101.30.1") },
+            { 9.0,  inet_address("192.102.30.1") },
+            { 10.0, inet_address("192.102.40.1") },
+            { 11.0, inet_address("192.102.40.2") }
+    };
+
+    std::vector<unsigned> nodes_shard_count(ring_points.size(), 3);
+
+    locator::token_metadata::config tm_cfg;
+    tm_cfg.topo_cfg.this_endpoint = ring_points[0].host;
+    tm_cfg.topo_cfg.local_dc_rack = { snitch.local()->get_datacenter(), snitch.local()->get_rack() };
+    tm_cfg.topo_cfg.this_host_id = ring_points[0].id;
+    locator::shared_token_metadata stm([] () noexcept { return db::schema_tables::hold_merge_lock(); }, tm_cfg);
+
+    // Initialize the token_metadata
+    stm.mutate_token_metadata([&] (token_metadata& tm) -> future<> {
+        auto& topo = tm.get_topology();
+        for (const auto& [ring_point, endpoint, id] : ring_points) {
+            std::unordered_set<token> tokens;
+            tokens.insert({dht::token::kind::key, d2t(ring_point / ring_points.size())});
+            topo.add_node(id, endpoint, make_endpoint_dc_rack(endpoint), locator::node::state::normal, 1);
+            tm.update_host_id(id, endpoint);
+            co_await tm.update_normal_tokens(std::move(tokens), id);
+        }
+    }).get();
+
+    std::map<sstring, sstring> options323 = {
+        {"100", "3"},
+        {"101", "2"},
+        {"102", "3"}
+    };
+    locator::replication_strategy_params params323(options323, 8);
+
+    auto ars_ptr = abstract_replication_strategy::create_replication_strategy(
+        "NetworkTopologyStrategy", params323);
+
+    auto tab_awr_ptr = ars_ptr->maybe_as_tablet_aware();
+    BOOST_REQUIRE(tab_awr_ptr);
+
+    auto s = schema_builder("ks", "tb")
+        .with_column("pk", utf8_type, column_kind::partition_key)
+        .with_column("v", utf8_type)
+        .build();
+
+    std::unordered_map<sstring, size_t> const& new_dc_rep_factor = {
+        {"100", 3},
```

Comment:
Just occurred to me that we store both the DC and the RF as `sstring`, you do it above as well https://github.com/scylladb/scylladb/pull/17116/files#diff-d903c1883207b8d23c27a9835c250a0788ee6e4668a73fae3378b0e5ef6e377aR1812, I think we should keep this convention, so your `reallocate_tablets_for_new_rf` would take `std::map<sstring, sstring>`, convert it to sstring:size_t and later convert to sstring:sstring back.